src/main/java/uk/gov/hmcts/reform/civil/service/citizen/UpdateCaseManagementDetailsService.java

```diff
@@ -0,0 +1,118 @@
+            .ifPresent(dq -> Optional.ofNullable(dq.getApplicant1DQRequestedCourt())
+                .ifPresent(requestedCourt -> builder.applicant1DQ(
+                    dq.toBuilder().applicant1DQRequestedCourt(correctCaseLocation(requestedCourt, availableLocations))
+                        .build())));
+    }
+
+    private void updateRespondent1RequestedCourtDetails(CaseData caseData, CaseData.CaseDataBuilder<?, ?> builder, List<LocationRefData> availableLocations) {
+        Optional.ofNullable(caseData.getRespondent1DQ())
+            .ifPresent(dq -> Optional.ofNullable(dq.getRespondent1DQRequestedCourt())
+                .ifPresent(requestedCourt -> builder.respondent1DQ(
+                    dq.toBuilder().respondent1DQRequestedCourt(correctCaseLocation(requestedCourt, availableLocations))
+                        .build())));
+    }
+
+    private RequestedCourt correctCaseLocation(RequestedCourt requestedCourt, List<LocationRefData> locations) {
+        String locationLabel = requestedCourt.getCaseLocation().getBaseLocation();
+        LocationRefData preferredLocation = locations.stream()
+            .filter(locationRefData -> courtLocationUtils.checkLocation(locationRefData, locationLabel))
+            .findFirst().orElseThrow(RuntimeException::new);
+        return requestedCourt.toBuilder()
+            .responseCourtCode(preferredLocation.getCourtLocationCode())
+            .caseLocation(LocationHelper.buildCaseLocation(preferredLocation))
+            .build();
+    }
+
+    private List<LocationRefData> fetchLocationData(CallbackParams callbackParams) {
+        String authToken = callbackParams.getParams().get(BEARER_TOKEN).toString();
+        return locationRefDataService.getCourtLocationsForDefaultJudgments(authToken);
+    }
+
+    private StringBuilder caseParticipants(CaseData caseData) {
+        StringBuilder participantString = new StringBuilder();
+        MultiPartyScenario multiPartyScenario = getMultiPartyScenario(caseData);
+        if (multiPartyScenario.equals(MultiPartyScenario.ONE_V_TWO_ONE_LEGAL_REP)
+            || multiPartyScenario.equals(MultiPartyScenario.ONE_V_TWO_TWO_LEGAL_REP)) {
+            participantString.append(caseData.getApplicant1().getPartyName())
+                .append(" v ").append(caseData.getRespondent1().getPartyName())
+                .append(" and ").append(caseData.getRespondent2().getPartyName());
+
+        } else if (multiPartyScenario.equals(MultiPartyScenario.TWO_V_ONE)) {
+            participantString.append(caseData.getApplicant1().getPartyName())
+                .append(" and ").append(caseData.getApplicant2().getPartyName()).append(" v ")
+                .append(caseData.getRespondent1()
+                            .getPartyName());
+
+        } else {
+            participantString.append(caseData.getApplicant1().getPartyName()).append(" v ")
+                .append(caseData.getRespondent1()
+                            .getPartyName());
```

Comment:
One liner `caseData.getRespondent1().getPartyName()` instead of 
```caseData.getRespondent1()```
```.getPartyName()```

Reply:
done